src/vendor/core/warning.js

```diff
@@ -31,6 +31,10 @@ if (__DEV__) {
       );
     }
 
+    if (format.indexOf('Failed CompositeComponent proptype check. ') === 0) {
+      return; // Ignore CompositeComponent proptype check.
+    }
+
```

Comment:
There are two places where proptype errors are detected/thrown.  The first is on the Element and the second is on the Component.  In open source, we now warn on the Element, but internally we still warn on the component and disable the element warning (while we fix our shit internally).  The intent here is that the warnings fired by the CompositeComponent shouldn't actually happen since they should be fired by the Element, so the warning module just silences those warnings.  If your proptypes are wrong and you're not getting any warnings, that sounds like an issue.  If you can create a simple demo on jsfiddle, we can take a look into it.